proposals/open-telemetry/microprofile-opentelemetry-tracing.asciidoc

```diff
@@ -0,0 +1,193 @@
+annotation. This will create a new Span and establish any required relationships with the current Trace context.
+
+Method parameters can be annotated with the `io.opentelemetry.extension.annotations.SpanAttribute` annotation to
+indicate which method parameters should be part of the Trace.
+
+Example:
+[source,java]
+----
+@ApplicationScoped
+class SpanBean {
+    @WithSpan
+    void span() {
+
+    }
+
+    @WithSpan("name")
+    void spanName() {
+
+    }
+
+    @WithSpan(kind = SERVER)
+    void spanKind() {
+
+    }
+
+    @WithSpan
+    void spanArgs(@SpanAttribute(value = "arg") String arg) {
+
+    }
+}
+----
+
+==== Obtain a SpanBuilder
+
+By obtaining a `SpanBuilder` from the current `Tracer` and calling
+`io.opentelemetry.api.trace.Tracer.spanBuilder(String)`. In this case, it is the developer's responsibility to ensure
+that the `Span` is properly created, closed, and propagated.
+
+Example:
+[source,java]
+----
+@RequestScoped
+@Path("/")
+public class SpanResource {
+    @Inject
+    Tracer tracer;
+
+    @GET
+    @Path("/span/new")
+    public Response spanNew() {
+        Span span = tracer.spanBuilder("span.new")
+                .setSpanKind(INTERNAL)
+                .setParent(Context.current().with(this.span))
+                .setAttribute("my.attribute", "value")
+                .startSpan();
+
+        span.end();
+
+        return Response.ok().build();
+    }
+}
+----
+
+NOTE: Start and end a new `Span` will add a child `Span` to the current one enlisted by the automatic instrumentation
+of Jakarta REST Applications.
+
+==== Obtain the current Span
+
+By obtaining the current `Span` to add attributes. The Span lifecycle is managed by the implementation.
+
+Example:
+[source,java]
+----
+@RequestScoped
+@Path("/")
+public class SpanResource {
+    @Inject
+    Span span;
+
+    @GET
+    @Path("/span/current")
+    public Response spanCurrent() {
+        span.setAttribute("my.attribute", "value");
+        return Response.ok().build();
+    }
+}
+----
+
+=== Agent Instrumentation
+
+TODO
+
+=== Access to the OpenTelemetry Tracing API
+
+The implemention of MicroProfile OpenTelemetry must provide the following CDI beans for supporting contextual instance injection:
+
+- `io.opentelemetry.api.OpenTelemetry`
+- `io.opentelemetry.api.baggage.Baggage`
+- `io.opentelemetry.api.trace.Span`
+- `io.opentelemetry.api.trace.Tracer`
+
+=== Configuration
+
+OpenTelemetry must be configured by MicroProfile Config following the configuration properties detailed in:
+
+- https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure[OpenTelemetry SDK Autoconfigure]
+(excluding properties related to Metrics).
+- https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/manual-instrumentation.md[Manual Instrumentation]
+
+An implementation may opt to not support a subset of configuration properties related to a specific configuration. For
```

Comment:
```suggestion
An implementation may opt to not support a subset of configuration properties related to a specific configuration. However, a bridge between MP Config and the [DefaultConfigProperties ](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/DefaultConfigProperties.java) must be provided. For
```

Reply:
I'm not sure if we should go into the implementation details. Of course, this is the obvious way to do it, but maybe other implementations may want to do it differently, and this may be a limiting factor.

Reply:
Otherwise how do you suggest we bridge the configs of both worlds? I think we need to be more precise

Reply:
I think adding `todo` to evaluate the bridge between MP Config and OTel Config might be a way forward at this stage.

Reply:
For instance, the default OTel implementation also reads values from `System.getProperties`, so another possible way to implement is to iterate the MP Config keys on init and call a `System.setProperty`. Yes, in a way you have a bridge, but an indirect one. 

It is ok to say that a bridge is required, I just want to avoid referencing OTel classes directly into the spec.

Reply:
sounds good to me just saying the bridge is required